aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -76,99 +139,533 @@ void filterEngineConfigs(
 
 struct CacheKey {
   ConvolutionParams params;
-  uint8_t input_alignment;
-  uint8_t weight_alignment;
-  uint8_t output_alignment;
+  cudnnBackendDescriptorType_t operation;
+  uint8_t x_alignment;
+  uint8_t w_alignment;
+  uint8_t y_alignment;
 };
 
-// FIXME: make this thread-safe by reusing the benchmark cache in Conv_v7.cpp
-std::unordered_map<CacheKey, cudnn_frontend::ManagedOpaqueDescriptor, ParamsHash<CacheKey>, ParamsEqual<CacheKey>> engine_cache;
+struct CacheKeyFused {
+  ConvolutionParams params;
```

Comment:
is device now part of convolutionParams? I remember we had to add it to cudnn caching key somewhere, cc @xwang233 @zasdfgbnm

Reply:
should be here: https://github.com/pytorch/pytorch/blob/e6ef87001c2b7df1eac139adcaef174395e3a5a2/aten/src/ATen/native/cudnn/ConvShared.h#L20